### PR TITLE
feat: Add note about available update

### LINKED ISSUE #325: Add note about available update
It would be good to add subtle note informing about available update.

### PR DESCRIPTION
**Pull request by pktiuk**
Closes #325 

TODO:

- check why on windows received json is empty
- Add proper handling in case of wrong json content https://stackoverflow.com/questions/17686561/qt-qnetworkreply-is-always-empty

### CODE REVIEW

**Comment by pktiuk** [comment]
obraz

**Comment by pktiuk** [comment]
@gombosg @avinal 

What do you think about this way of showing update notification to users?

**Comment by gombosg** [comment]
Hey it's good, I like it. It's not sending any data (GET request), so it's OK from a privacy perspective. Also it looks safe because github.com is hardcoded in it.

Later on it would be great to have a 'check for updates on startup' option in the options page so that users can disable the feature. I think that users might expect that but it's not a blocker, especially that privacy-minded users can just compile their own version.

Also, what do you think about disabling this in repository and flatpak versions, where users are unable to update by themselves?

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -133,7 +133,7 @@ jobs:
                   install-deps: "true"
 
             - name: Configure CMake
-              run: cmake -DCMAKE_BUILD_TYPE=Release -DANTIMICROX_PKG_VERSION="GitHub Windows Release" -B '${{ github.workspace }}'/build
+              run: cmake -DCMAKE_BUILD_TYPE=Release -DCHECK_FOR_UPDATES=ON -DANTIMICROX_PKG_VERSION="GitHub Windows Release" -B '${{ github.workspace }}'/build
```
Comment (pktiuk) [inline .github/workflows/release.yml:136]: @gombosg 
> Also, what do you think about disabling this in repository and flatpak versions, where users are unable to update by themselves? 

This is disabled by default. Only Windows build uses this feature

Comment (gombosg) [reply .github/workflows/release.yml:136]: Sounds cool then! Window$ users need some handholding 🤡

**Comment by pktiuk** [comment]
For Widows build I get warning:
`QSslSocket::connectToHostEncrypted: TLS initialization failed`

**Comment by pktiuk** [comment]
Maybe I will have to add package https://packages.msys2.org/package/libopenssl for windows and bundle more libs.

**Comment by pktiuk** [comment]
This is OpenSSL issue.  
After instaling library from https://slproweb.com/products/Win32OpenSSL.html everything works.  
Windows makes simple tasks a really tedious...

**Pull request merged**